WalletWasabi.Gui/Controls/QrCode.cs

```diff
@@ -30,6 +32,42 @@ public QrCode()
 				.Where(x => x != null)
 				.ObserveOn(RxApp.MainThreadScheduler)
 				.Subscribe(x => FinalMatrix = AddPaddingToMatrix(x));
+
+			this.WhenAnyValue(x => x.QRImageSavePath)
+				.Where(x => !string.IsNullOrWhiteSpace(x) || !string.IsNullOrEmpty(x))
+				.Where(x => FinalMatrix != null)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(x => GenerateQRCodeBitmap(x));
+		}
+
+		private void GenerateQRCodeBitmap(string x)
+		{
+			try
+			{
+				var pixSize = PixelSize.FromSize(CoercedSize, 1);
+				using var rtb = new RenderTargetBitmap(pixSize);
+
+				rtb.Render(this);
+				rtb.Save(x);
+			}
+			catch (Exception ex)
+			{
+				Logger.LogDebug(ex);
+			}
+		}
+
+		public static readonly DirectProperty<QrCode, string> QRImageSavePathProperty =
```

Comment:
As I mentioned before. Do we need this?

Reply:
I think our idea for that was to decouple the saving function to the viewmodel since it requires framework-side code (dialogs and stuff). hence me & @danwalmsley decided to make an indirection